tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8013,6 +8013,54 @@ def err_hlsl_reorder_unsupported_stage : Error<
 def err_hlsl_hitobject_unsupported_stage : Error<
    "dx::HitObject is unavailable in shader stage '%0' (requires 'raygeneration', 'closesthit' or 'miss')">;
+
+// Linear Algebra Operations
+def err_hlsl_linalg_param_must_be_const : Error<
+  "'%0' must be a constant parameter">;
+def err_hlsl_linalg_vector_incorrect_type : Error<
+  "%0 is incorrect type, must be 16-bit or 32-bit 'unsigned int', "
+  "'signed int' or 'float'">;
+def err_hlsl_linalg_isunsigned_incorrect_for_given_type : Error<
+  "%0 must be %select{false|true}1 for a %2 vector type">;
+def err_hlsl_linalg_interpretation_value_incorrect : Error<
+  "%0 is an invalid %select{Memory|Register}1 Interpretation value">;
+def err_hlsl_linalg_matrix_layout_is_not_transposable : Error<
+  "RowMajor and ColumnMajor matrices are not transposable">;
+def err_hlsl_linalg_optimal_matrix_layout_matrix_stride_must_be_zero : Error<
+  "for optimal matrix layout, matrix stride must be zero">;
+def err_hlsl_linalg_matrix_dim_must_not_be_zero: Error<
+  "matrix dimension must not be zero">;
+def err_hlsl_linalg_matrix_layout_invalid : Error<
+  "matrix layout %0 is not valid, must be in the range %1 - %2">;
+def err_hlsl_linalg_function_requires_shader_model_6_9_or_above : Error<
+  "intrinsic function %0 requires shader model 6.9 or greater">;
+
+def err_hlsl_linalg_mul_muladd_output_vector_size_not_equal_to_matrix_M : Error<
+  "output vector length must be equal to Matrix M dimension in a linalg Mul/MulAdd operation">;
+def err_hlsl_linalg_mul_muladd_unpacked_input_vector_size_not_equal_to_matrix_K : Error<
+  "unpacked input vector length must be equal to Matrix K dimension in a linalg Mul/MulAdd operation">;
+def err_hlsl_linalg_mul_muladd_packed_input_vector_size_incorrect : Error<
+  "packed input vector length must be the smallest number that can hold matrix dim K values of the "
+  "packed(smaller) type in linalg mul/muladd operations">;
+def err_hlsl_linalg_mul_muladd_isUnsigned_for_packed_input_must_be_true : Error<
+  "IsInputUnsigned must be true for packed input interpretations in linalg mul/muladd operations">;
+def err_hlsl_linalg_mul_muladd_packed_input_vector_must_be_uint : Error<
+  "packed input vector type must be a 32-bit unsigned int in linalg mul/muladd operations, "
+  "packed formats uint8_t4_packed and sint8_t4_packed are not supported currently">;
+def err_hlsl_linalg_mul_muladd_exceeds_max_matrix_dim_m: Error<
+  "matrix dimension M must be less than %0, in a linalg Mul/MulAdd operation">;
+def err_hlsl_linalg_mul_muladd_exceeds_max_matrix_dim_k_unpacked_input: Error<
+  "matrix dimension K when using unpacked input vectors must be less than %0, in a "
+  "linalg Mul/MulAdd operation">;
+def err_hlsl_linalg_mul_muladd_exceeds_max_matrix_dim_k_packed_input: Error<
+  "matrix dimension K when using packed input vectors must be less than %0, in a "
+  "linalg Mul/MulAdd operation">;
```

Comment:
We can collapse these down to a single error (this makes for more compact binary representation.
```suggestion
def err_hlsl_linalg_mul_muladd_invalid_dim: Error<
  "matrix dimension %select{M|K when using unpacked input vectors|K "
  "when using packed input vectors}0 must be less than %1, in a linalg "
  "Mul/MulAdd operation">;
```